client/modules/IDE/components/Header/Nav.jsx

```diff
@@ -1,414 +1,318 @@
-              {this.props.t('Nav.SignUp')}
-            </span>
-          </Link>
-        </li>
-      </ul>
-    );
-  }
+  // TODO: use selectRootFile selector
+  const rootFile = useSelector(
+    (state) => state.files.filter((file) => file.name === 'root')[0]
+  );
 
-  renderAuthenticatedUserMenu() {
-    return (
-      <ul className="nav__items-right" title="user-menu">
-        {getConfig('TRANSLATIONS_ENABLED') && this.renderLanguageMenu()}
-        <NavDropdownMenu
-          id="account"
-          title={
-            <span>
-              {this.props.t('Nav.Auth.Hello')}, {this.props.user.username}!
-            </span>
-          }
-        >
-          <NavMenuItem href={`/${this.props.user.username}/sketches`}>
-            {this.props.t('Nav.Auth.MySketches')}
-          </NavMenuItem>
-          <NavMenuItem
-            href={`/${this.props.user.username}/collections`}
-            hideIf={!getConfig('UI_COLLECTIONS_ENABLED')}
-          >
-            {this.props.t('Nav.Auth.MyCollections')}
-          </NavMenuItem>
-          <NavMenuItem href={`/${this.props.user.username}/assets`}>
-            {this.props.t('Nav.Auth.MyAssets')}
-          </NavMenuItem>
-          <NavMenuItem href="/account">
-            {this.props.t('Preferences.Settings')}
-          </NavMenuItem>
-          <NavMenuItem onClick={this.props.logoutUser}>
-            {this.props.t('Nav.Auth.LogOut')}
-          </NavMenuItem>
-        </NavDropdownMenu>
-      </ul>
-    );
-  }
+  const cmRef = useContext(CmControllerContext);
 
-  renderUserMenu() {
-    const isLoginEnabled = getConfig('LOGIN_ENABLED');
-    const isAuthenticated = this.props.user.authenticated;
+  const dispatch = useDispatch();
 
-    if (isLoginEnabled && isAuthenticated) {
-      return this.renderAuthenticatedUserMenu();
-    } else if (isLoginEnabled && !isAuthenticated) {
-      return this.renderUnauthenticatedUserMenu();
-    }
+  const { t } = useTranslation();
+  const {
+    newSketch,
+    saveSketch,
+    downloadSketch,
+    shareSketch
+  } = useSketchActions();
 
-    return null;
-  }
+  const replaceCommand =
+    metaKey === 'Ctrl' ? `${metaKeyName}+H` : `${metaKeyName}+⌥+F`;
 
-  renderLeftLayout() {
-    switch (this.props.layout) {
-      case 'dashboard':
-        return this.renderDashboardMenu();
-      case 'project':
-      default:
-        return this.renderProjectMenu();
-    }
-  }
+  return (
+    <ul className="nav__items-left">
+      <li className="nav__item-logo">
+        <LogoIcon
+          role="img"
+          aria-label={t('Common.p5logoARIA')}
+          focusable="false"
+          className="svg__logo"
+        />
+      </li>
+      <NavDropdownMenu id="file" title={t('Nav.File.Title')}>
+        <NavMenuItem onClick={newSketch}>{t('Nav.File.New')}</NavMenuItem>
+        <NavMenuItem
+          hideIf={
+            !getConfig('LOGIN_ENABLED') || (project?.owner && !isUserOwner)
```

Comment:
FYI that second condition is the negative of `canEditProjectName`/`selectCanEditSketch`.